src/qibo/tests/test_derivative.py

```diff
@@ -52,6 +52,6 @@ def test_derivative(backend):
     assert isinstance(test_hamiltonian, hamiltonians.AbstractHamiltonian)
     # check of known values
```

Comment:
It is not really needed.

Reply:
Thank you very much @DiegoGM91 for your review. 
I proceed with these modifications.

I purpose to open a new PR, because this one is starting to get pretty big.
I have already opened a second PR related to the RDP: the #735. 
I'm going to add also you as reviewer.